### ISSUE TITLE
Deprecated on php8.1

### ISSUE THREAD

**Issue by tpk109** [scssphp/scssphp#651]
Deprecated: Return type of ScssPhp\ScssPhp\Node\Number::offsetExists($offset) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/scssphp/scssphp/src/Node/Number.php on line 146 Deprecated: Return type of ScssPhp\ScssPhp\Node\Number::offsetGet($offset) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/scssphp/scssphp/src/Node/Number.php on line 171 Deprecated: Return type of ScssPhp\ScssPhp\Node\Number::offsetSet($offset, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/scssphp/scssphp/src/Node/Number.php on line 197 Deprecated: Return type of ScssPhp\ScssPhp\Node\Number::offsetUnset($offset) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /vendor/scssphp/scssphp/src/Node/Number.php on line 205

**Comment by stof** [comment]
which version of scssphp are you using ?

**Comment by bymem** [comment]
Same problem running vers 1.11.0

**Comment by stof** [comment]
@bymem the 1.11.0 version **has** the `ReturnTypeWillChange` attribute on this method: https://github.com/scssphp/scssphp/blob/33749d12c2569bb24071f94e9af828662dabb068/src/Node/Number.php#L152

So please check that you actually use 1.11.0 (and without weird patches applied on it that would remove the attribute).

**Comment by stof** [comment]
Based on the line number given in the error message (mentioning that `ScssPhp\ScssPhp\Node\Number::offsetExists` is defined on line 146), it looks like version 1.4.1 or older is used (in 1.5.0, changes in the file pushed that method to line 152).
Version 1.4.1 was released in January 2021, so before the PHP 8.1 stable release (and probably even before that deprecation was implemented in the PHP dev branch). We have made 15 releases after that one.
The attribute is present since version 1.8.1.
